File: coding/12th October 2025/Two Sum.java
//Two Sum
class Solution
{
    public int[] twoSum(int[] nums, int target)
    {
        int[] result = new int[2];
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++)
        {
            if(map.containsKey(target - nums[i]))
            {
                result[0] = map.get(target - nums[i]);
                result[1] = i;
                break;
            }
            map.put(nums[i], i);
        }
        return result;
    }
}
/*
class Solution
{
    public int[] twoSum(int[] nums, int target)
    {
        int[] res = new int[2];
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++)
        {
            if(map.containsKey(target - nums[i]) && i != map.get(target - nums[i]))
            {
                res[0] = i;
                res[1] = map.get(target - nums[i]);
                break;
            }
            map.put(nums[i], i);
        }
        return res;
    }
}
*/

File: Coding in Python/04th January 2026/Two Sum.py
#Two Sum
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = [0] * 2
        mp = {}
        for i, ele in enumerate(nums):
            if target - ele in mp:
                res[0] = mp.get(target - ele)
                res[1] = i
                break
            mp[ele] = i
        return res
